packages/compass-connections/src/hooks/use-connection-supports.ts

```diff
@@ -0,0 +1,44 @@
+import type { ConnectionState } from '../stores/connections-store-redux';
+import { useConnectionForId } from '../stores/store-context';
+
+// only one for now
+type ConnectionFeature = 'rollingIndexCreation';
+
+function isFreeOrSharedTierCluster(instanceSize: string | undefined): boolean {
+  if (!instanceSize) {
+    return false;
+  }
+
+  return ['M0', 'M2', 'M5'].includes(instanceSize);
+}
+
+function supportsRollingIndexCreation(connection: ConnectionState) {
+  const atlasMetadata = connection.info?.atlasMetadata;
+
+  if (!atlasMetadata) {
+    return false;
+  }
+
+  const { clusterType, instanceSize } = atlasMetadata;
+  return (
+    !isFreeOrSharedTierCluster(instanceSize) &&
+    (clusterType === 'cluster' || clusterType === 'replicaSet')
+  );
+}
+
+export function useConnectionSupports(
+  connectionId: string,
+  connectionFeature: ConnectionFeature
+): boolean {
+  const connection = useConnectionForId(connectionId);
+
+  if (!connection) {
+    return false;
+  }
+
+  if (connectionFeature === 'rollingIndexCreation') {
+    return supportsRollingIndexCreation(connection);
+  }
+
+  return false;
```

Comment:
So what I meant here when I mentioned writing this with `useSelector` is that instead of using `useConnectionForId` (or any other existing hook that will subscribe your component to the whole state of the connection: statuses, connection info, all of it, a lot of unrelated state), you should rather write it using `useSelector` directly and push as much of logic inside the selector function and make selector return the value you are actually interested in. That way you will make sure that no matter how many changes happen to the whole connection state, this hook will only re-render components using it when the actual value of whether or not the feature flag is supported is changed. This aligns with redux style guilde both for [using `connect` function](https://redux.js.org/style-guide#connect-more-components-to-read-data-from-the-store) and [using hooks](https://redux.js.org/style-guide#call-useselector-multiple-times-in-function-components):

> prefer calling useSelector many times and retrieving smaller amounts of data

So the implementation here can look something like this:

```suggestion
  return useSelector(state => {
    const connection = state.connections.byId[connectionId];
    
    if (!connection) {
      return false;
    }
    
    if (connectionFeature === 'rollingIndexCreation') {
      return supportsRollingIndexCreation(connection);
    }
    
    return false;
  })
```

Because `useSelector` returns a boolean, the change of that boolean is the only thing that would trigger a re-render of the component. So implementation is very similar to what you have, but at the same time very different in a pretty important way 😄

Reply:
Changed this. At first I didn't push all the changes. Fixed now :)